etl/steps/data/garden/growth/2024-01-04/gdp_historical.py

```diff
@@ -0,0 +1,154 @@
+"""Load a meadow dataset and create a garden dataset."""
+
+import owid.catalog.processing as pr
+from owid.catalog import Table
+
+from etl.data_helpers import geo
+from etl.helpers import PathFinder, create_dataset
+
+# Get paths and naming conventions for current step.
+paths = PathFinder(__file__)
+
+# Define constants: variables to process and references years where merge is done.
+VAR_LIST = ["gdp", "gdp_per_capita"]
+YEAR_WDI_MPD = 1990
+YEAR_MPD_MD = 1820
+
+# Define accuracy of data (in tens)
+# 6 means that the data is accurate to the 10^6
+ACCURACY_GDP = 6
+ACCURACY_GDP_PER_CAPITA = 2
+
+
+def run(dest_dir: str) -> None:
+    #
+    # Load inputs.
+    #
+    # Load dataset and tables
+
+    # World Bank WDI
+    ds_wdi = paths.load_dataset("wdi")
+    tb_wdi = ds_wdi["wdi"].reset_index()
+
+    # Maddison Project Database
+    ds_mpd = paths.load_dataset("ggdc_maddison")
+    tb_mpd = ds_mpd["maddison_gdp"].reset_index()
+
+    # Maddison Database
+    ds_md = paths.load_dataset("maddison_database")
+    tb_md = ds_md["maddison_database"].reset_index()
+
+    #
+    # Process data.
+    tb = process_and_combine_datasets(tb_wdi, tb_mpd, tb_md)
+
+    tb = tb.set_index(["country", "year"], verify_integrity=True)
+
+    #
+    # Save outputs.
+    #
+    # Create a new garden dataset with the same metadata as the meadow dataset.
+    ds_garden = create_dataset(
```

Comment:
Yes, I am combining variables with different units, but I applied the growth of one of them retroactively to the other variable to keep consistency and to avoid mixing different things. The ETL (because of the metadata) still thinks I am proceeding wrong, but it's not the case. I hope the changes are clearer now.